plugins/languages/rust/crates.nix

```diff
@@ -0,0 +1,115 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "crates";
+    originalName = "crates.nvim";
+    defaultPackage = pkgs.vimPlugins.crates-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    settingsOptions = {
+      on_attach = helpers.defaultNullOpts.mkLuaFn "function(bufnr) end" ''
+        `fun(bufnr: integer)`
+
+        Callback to run when a `Cargo.toml` file is opened.
+
+        NOTE: Ignored if `autoload` is disabled.
+      '';
+
+      cmp = {
+        enabled = helpers.defaultNullOpts.mkBool false ''
+          Whether to load and register the `nvim-cmp` source.
+
+          NOTE: Ignored if `autoload` is disabled.
+          You may manually register it, after `nvim-cmp` has been loaded.
+          ```lua
+            require("crates.src.cmp").setup()
+          ```
+        '';
+      };
+
+      null_ls = {
+        enabled = helpers.defaultNullOpts.mkBool false ''
+          Whether to register the `null-ls.nvim` source.
+        '';
+
+        name = helpers.defaultNullOpts.mkStr "crates.nvim" ''
+          The `null-ls.nvim` name.
+        '';
+      };
+
+      lsp = {
+        enabled = helpers.defaultNullOpts.mkBool false ''
+          Whether to enable the in-process language server.
+        '';
+
+        name = helpers.defaultNullOpts.mkStr "crates.nvim" ''
+          The lsp server name.
+        '';
+
+        on_attach = helpers.defaultNullOpts.mkLuaFn "function(client, bufnr) end" ''
+          `fun(client: lsp.Client, bufnr: integer)`
+
+          Callback to run when the in-process language server attaches to a buffer.
+
+          NOTE: Ignored if `autoload` is disabled.
+        '';
+
+        # on_attach = helpers.defaultNullOpts.mkLuaFn "__lspOnAttach" ''
+        # '';
```

Comment:
This should not be present (And like mentionned we should do something about setting the lsp default value to __lspOnAttach to benefit from the other options, like format.nvim, or LSP specific mappings that we could introduce to solve  #1219))